openstudiocore/src/model/FileOperations.cpp

```diff
@@ -79,242 +79,69 @@
 #include "../utilities/core/Logger.hpp"
 #include "../utilities/core/PathHelpers.hpp"
 #include "../utilities/core/Assert.hpp"
+#include "../utilities/core/FilesystemHelpers.hpp"
 
 #include <OpenStudio.hxx>
 
-#include <QDir>
+#include <QFile>
+#include <QFileInfo>
 #include <QMessageBox>
-#include <QTemporaryFile>
-
-
 
 
 namespace openstudio {
 namespace model {
-
-  bool removeDir(const QString &dirName)
+  bool copyDir(const openstudio::path &sourceDir, const openstudio::path &destinationDir)
   {
-    bool result = true;
-
-    LOG_FREE(Info, "removeDir", "removeDir '" << toString(dirName) << "'");
-
-    QDir dir(dirName);
-    if (dir.exists(dirName)) {
-      bool test = true;
-
-      for (const QFileInfo& info : dir.entryInfoList(QDir::NoDotAndDotDot | QDir::System | QDir::Hidden | QDir::AllDirs | QDir::Files, QDir::DirsFirst))
-      {
-        if (info.isDir()) {
-          test = removeDir(info.absoluteFilePath());
-          if (!test){
-            LOG_FREE(Error, "removeDir", "Failed to remove directory '" << toString(info.absoluteFilePath()) << "'");
-            result = false;
-
-            // DLM: do we really want to give up here?
-            //return result;
-          }
-        } else{
-          test = QFile::remove(info.absoluteFilePath());
-          if (!test){
-            LOG_FREE(Error, "removeDir", "Failed to remove file '" << toString(info.absoluteFilePath()) << "'");
-            result = false;
-
-            // DLM: do we really want to give up here?
-            //return result;
-          }
-        }
-      }
-      test = dir.rmdir(dirName);
-      if (!test){
-        LOG_FREE(Error, "removeDir", "Failed to remove directory '" << toString(dirName) << "'");
-        result = false;
-
-        // DLM: do we really want to give up here?
-        //return result;
-      }
-    }
-
-    return result;
-  }
-
-  bool synchDirStructures(const QString &srcPath, const QString &dstPath)
-  {
-    bool result = true;
-    bool test = true;
-
-    QDir srcDir(srcPath);
-    QDir dstDir(dstPath);
-
-    if (srcDir.canonicalPath() == dstDir.canonicalPath()){
```

Comment:
We're not handling this case anymore?

Reply:
In general the old behavior was too complicated for my tastes, but I'm not sure how we evolved to get there.

Reply:
Well, then do we want simplified behavior for today and see if the old behavior *needs* to exist, or do we go back and carefully re-create the old behavior? (I can at the very list make sure we're not copying onto ourselves as this code is checking for.)

Reply:
This is also resolved with 5d54887cf1990b6230771ac05e76f56ebfd60286